site/content/extend/plugins/webapp/proper-configuration.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Proper Configuration
```

Comment:
How do item 2 and 3 fit under this title?

Reply:
Agreed, they feel more like plugin architectural concerns vs. "how to develop on plugins".

Reply:
@hanzei @lieut-data As noted in the PR description, "Proper Configuration" is just a placeholder to get this PR reviewed. It's certainly not the right phrase to use. Do you have any advice on where tab/page it should go in the docs, or what the page should be called if it's a new page?

I don't think the webapp design best practices page is right, because that's dedicated to information about adding components to different parts of the interface.

Reply:
@hanzei @lieut-data How do you think the docs should be structured here, regarding "making sure a plugin's HTTP requests to the server have the correct URL/headers?"

Reply:
There is a "Best practices" document in https://developers.mattermost.com/extend/plugins/webapp/best-practices/. While is currently design focused, it could be instead contain both code and design practices.

Reply:
@hanzei I've moved the architectural concern docs to the best practices page